src/main/java/com/ohdab/member/service/AddTeacherService.java

```diff
@@ -0,0 +1,52 @@
+package com.ohdab.member.service;
+
+import com.ohdab.member.exception.NoMemberException;
+import com.ohdab.member.repository.MemberRepository;
+import com.ohdab.member.service.dto.AddTeacherReqDto;
+import com.ohdab.member.service.dto.JoinReqDto;
+import com.ohdab.member.service.helper.MemberHelperService;
+import com.ohdab.member.service.usecase.AddTeacherUsecase;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class AddTeacherService implements AddTeacherUsecase {
+
+    private final MemberRepository memberRepository;
+    // private final JoinService joinService;
+    private final MemberHelperService memberHelperService;
+
+    @Override
+    public void addTeacher(AddTeacherReqDto addTeacherReqDto) {
+        String name = addTeacherReqDto.getName();
+        name = changeNameIfDuplicated(name);
+        String password = name;
+        JoinReqDto joinReqDto =
+                JoinReqDto.builder()
+                        .name(name)
+                        .password(password)
+                        .role(List.of("TEACHER", "STUDENT"))
+                        .build();
+        // TODO:선생님 추가 시 회원가입 요청
+        // joinService.join(joinReqDto);
+        // throwIfJoinFailed(name);
+    }
+
+    private String changeNameIfDuplicated(String name) {
+        if (memberHelperService.checkIfMemberExistByName(memberRepository, name)) {
+            Long sameNameCount = memberRepository.countByMemberInfoNameContaining(name);
+            return name = name + sameNameCount.toString();
+        }
+        return name;
+    }
```

Comment:
```suggestion
    private String changeNameIfDuplicated(String name) {
        if (memberHelperService.checkIfMemberExistByName(memberRepository, name)) {
            long sameNameCount = memberRepository.countByMemberInfoNameContaining(name);
            return name = name + sameNameCount.toString();
        }
        return name;
    }
```
count값이 0 이상이라 not-null이 보장될거 같아서 원시타입으로 쓰는게 맞을거같은데

Reply:
화긴